org.metaborg.spoofax.meta.core/src/main/java/org/metaborg/spoofax/meta/core/pluto/build/StrIncrFrontEnd.java

```diff
@@ -0,0 +1,165 @@
+package org.metaborg.spoofax.meta.core.pluto.build;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import org.apache.commons.vfs2.FileObject;
+import org.metaborg.core.MetaborgException;
+import org.metaborg.core.MetaborgRuntimeException;
+import org.metaborg.core.action.EndNamedGoal;
+import org.metaborg.core.build.BuildInput;
+import org.metaborg.core.build.BuildInputBuilder;
+import org.metaborg.core.language.ILanguageImpl;
+import org.metaborg.spoofax.core.build.ISpoofaxBuildOutput;
+import org.metaborg.spoofax.core.build.SpoofaxCommonPaths;
+import org.metaborg.spoofax.core.unit.ISpoofaxTransformUnit;
+import org.metaborg.spoofax.meta.core.pluto.SpoofaxBuilder;
+import org.metaborg.spoofax.meta.core.pluto.SpoofaxBuilderFactory;
+import org.metaborg.spoofax.meta.core.pluto.SpoofaxBuilderFactoryFactory;
+import org.metaborg.spoofax.meta.core.pluto.SpoofaxContext;
+import org.metaborg.spoofax.meta.core.pluto.SpoofaxInput;
+import org.slf4j.helpers.MessageFormatter;
+import org.spoofax.interpreter.core.Tools;
+import org.spoofax.interpreter.terms.IStrategoList;
+import org.spoofax.interpreter.terms.IStrategoTerm;
+import org.sugarj.common.FileCommands;
+
+import com.google.common.collect.Iterables;
+
+import build.pluto.BuildUnit.State;
+import build.pluto.dependency.Origin;
+import io.usethesource.capsule.BinaryRelation;
+
+public class StrIncrFrontEnd extends SpoofaxBuilder<StrIncrFrontEnd.Input, StrIncrFrontEnd.Output> {
+    private static final String COMPILE_GOAL_NAME = "Compile";
+    private static final String STRATEGO_LANG_NAME = "StrategoSugar";
+
+    public static class Input extends SpoofaxInput {
+        private static final long serialVersionUID = 1548589152421064400L;
+
+        public final SpoofaxCommonPaths paths;
+        public final File inputFile;
+        public final Origin origin;
+
+        public Input(SpoofaxContext context, SpoofaxCommonPaths paths, File inputFile, Origin origin) {
+            super(context);
+
+            this.paths = paths;
+            this.inputFile = inputFile;
+            this.origin = origin;
+        }
+    }
+
+    public static class Output implements build.pluto.output.Output {
+        private static final long serialVersionUID = 3808911543715367986L;
+
+        public final BuildRequest request;
```

Comment:
While it might be nice to return a `BuildRequest` for itself for use as origin in clients, this requires additional serialization/equality/hashCode overhead for every `StrIncrFrontEnd` call and should be avoided. Client should call the `request` or `origin` methods.